gcp/src/test/java/org/apache/iceberg/gcp/biglake/BigLakeTableOperationsTest.java

```diff
@@ -0,0 +1,222 @@
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.exceptions.CommitFailedException;
+import org.apache.iceberg.exceptions.NoSuchTableException;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.types.Types;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+public class BigLakeTableOperationsTest {
+
+  @Rule public final MockitoRule mockito = MockitoJUnit.rule();
+  @Rule public TemporaryFolder tempFolder = new TemporaryFolder();
+
+  private static final String GCP_PROJECT = "my-project";
+  private static final String GCP_REGION = "us";
+  private static final String CATALOG_ID = "biglake";
+  private static final String DB_ID = "db";
+  private static final String TABLE_ID = "tbl";
+  private static final TableName TABLE_NAME =
+      TableName.of(GCP_PROJECT, GCP_REGION, CATALOG_ID, DB_ID, TABLE_ID);
+  private static final TableIdentifier SPARK_TABLE_ID = TableIdentifier.of(DB_ID, TABLE_ID);
+
+  @Mock private BigLakeClient bigLakeClient;
+
+  private BigLakeCatalog bigLakeCatalog;
+  private String warehouseLocation;
+  private BigLakeTableOperations tableOps;
+
+  @Before
+  public void before() throws Exception {
+    this.bigLakeCatalog = new BigLakeCatalog();
+    this.warehouseLocation = tempFolder.newFolder("hive-warehouse").toString();
+
+    bigLakeCatalog.setConf(new Configuration());
+    bigLakeCatalog.initialize(
+        CATALOG_ID,
+        /* properties= */ ImmutableMap.of(
+            BigLakeCatalog.PROPERTIES_KEY_GCP_PROJECT,
+            GCP_PROJECT,
+            CatalogProperties.WAREHOUSE_LOCATION,
+            warehouseLocation),
+        GCP_PROJECT,
+        GCP_REGION,
+        bigLakeClient);
+    this.tableOps = (BigLakeTableOperations) bigLakeCatalog.newTableOps(SPARK_TABLE_ID);
+  }
+
+  @Test
+  public void testDoFresh_fetchLatestMetadataFromBigLake() throws Exception {
+    Table createdTable = createTestTable();
+    reset(bigLakeClient);
+    when(bigLakeClient.getTable(TABLE_NAME)).thenReturn(createdTable);
+
+    tableOps.refresh();
+    assertEquals(
+        createdTable
+            .getHiveOptions()
+            .getParametersOrDefault(BigLakeTestUtils.METADATA_LOCATION_PROP, ""),
+        tableOps.currentMetadataLocation());
+
+    reset(bigLakeClient);
+    when(bigLakeClient.getTable(TABLE_NAME))
+        .thenThrow(new NoSuchTableException("error message getTable"));
+    // Refresh fails when table is not found but metadata already presents.
+    assertThrows(NoSuchTableException.class, () -> tableOps.refresh());
+  }
+
+  @Test
+  public void testDoFresh_failForNonIcebergTable() throws Exception {
```

Comment:
Non-Iceberg tables should result in `NoSuchIcebergTableException`, not refresh failures.

Reply:
Thanks. I updated the code to throw NoSuchIcebergTableException and updated the test as well.